TAScheduler/domain/CommandHandler.py

```diff
@@ -209,28 +209,30 @@ def _CreateLabHandler(self, cmd: [str]):
         return 'Lab created'
 
     def _AssignCourseTAHandler(self, cmd: [str]):
-        if self.currentUser is None or not self.currentUser.RoleIn(Role.Administrator, Role.Supervisor):
+        if self.currentUser is None or not self.currentUser.RoleIn(Role.Instructor, Role.Supervisor):
             return 'Must be logged in as an Administrator or Supervisor'
         if len(cmd) != 2:
             return ErrorMessages.INVALID_NUM_OF_ARGUMENTS
         c = Course.objects.filter(course_name=cmd[0]).first()
         if not c:
             return 'A course with that name does not exist'
+        if self.currentUser.RoleIn(Role.Instructor) and (c.instructor.act_email != self.currentUser.act_email):
+            return 'Instructors can only assign TAs to their own courses.'
         a = Account.objects.filter(act_email=cmd[1]).first()
         if not a or not a.RoleIn(Role.TA):
             return 'Only existing TAs can be assigned to the course'
         c.tas.add(a)
         return 'TA assigned to course'
 
     def _AssignCourseInstructorHandler(self, cmd: [str]):
-        if self.currentUser is None or not self.currentUser.RoleIn(Role.Administrator, Role.Supervisor):
-            return 'Must be logged in as an Administrator or Supervisor'
+        if self.currentUser is None or not self.currentUser.RoleIn(Role.Supervisor):
+            return 'Must be logged in as a Supervisor'
         if len(cmd) != 2:
             return ErrorMessages.INVALID_NUM_OF_ARGUMENTS
         try:
             c = Course.objects.get(course_name=cmd[0])
         except Exception:
-            return 'This course does not exist'
+            return 'A course with that name does not exist'
```

Comment:
Make sure if you're changing error messages that you also change tests to reflect them. Not sure if you've done it yet, just a general reminder to everyone to run "python manage.py tests" in terminal and verify all tests are passing before committing any PRs